Error: rpc error: code = Unavailable desc = transport is closing with datadog 2.19.1

Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version

0.14.4

### Affected Resource(s)

My terraform uses datadog_monitor and datadog_dashboard

### Terraform Configuration Files

```hcl
locals {
  team_name            = "tariffs"
  app_name             = "cost-projections"
  kafka_consumer_group = var.target_env == "prd" ? "cost-projections" : "cost-projections-${var.target_env}"
  slack_alerts_channel = var.target_env == "prd" ? "@slack-tariffs-prd-alerts" : "@slack-tariffs-test-alerts"

  kafka_monitors = {}

  kube_monitors = {}
}

# monitors/alerts
resource "datadog_monitor" "kafka_monitor" {
  for_each = local.kafka_monitors
  name     = "[${var.target_env}] [${local.team_name}] [${local.app_name}] ${each.value.title}"
  type     = "metric alert"
  message  = <<EOF
    {{#is_warning}}Has increased.{{/is_warning}}
    {{#is_alert}}Is currently too high.{{/is_alert}}
    {{#is_recovery}}Has recovered.{{/is_recovery}}
    ${local.slack_alerts_channel}
  EOF

  escalation_message = "Still too high."

  query               = "${each.value.window}:sum:kafka.consumer_lag{aiven-service:${each.value.service},consumer_group:${local.kafka_consumer_group},topic:${each.value.topic}} > ${each.value.critical_threshold}"
  notify_no_data      = false
  renotify_interval   = 180
  notify_audit        = false
  timeout_h           = 0
  include_tags        = true
  require_full_window = false
  new_host_delay      = 300
  evaluation_delay    = 300

  thresholds = {
    ok       = 0
    warning  = each.value.warning_threshold
    critical = each.value.critical_threshold
  }

  tags = ["team:${local.team_name}", "env:${var.target_env}", "app:${local.app_name}"]
}

resource "datadog_monitor" "kube_monitor" {
  for_each = local.kube_monitors
  name     = "[${var.target_env}] [${local.team_name}] [${local.app_name}] ${each.value.title}"
  type     = "metric alert"
  message  = <<EOF
    {{#is_alert}}Number of active pods has dropped below desired.{{/is_alert}}
    {{#is_recovery}}Has recovered.{{/is_recovery}}
    ${local.slack_alerts_channel}
  EOF

  escalation_message = each.value.escalation_message

  query = each.value.query

  notify_audit        = false
  notify_no_data      = true
  renotify_interval   = 0
  timeout_h           = 0
  include_tags        = true
  require_full_window = true
  new_host_delay      = 300
  evaluation_delay    = 300
  no_data_timeframe   = 10

  thresholds = {
    critical = each.value.critical_threshold
  }

  tags = ["team:${local.team_name}", "env:${var.target_env}", "app:${local.app_name}"]
}


# dashboard
resource "datadog_dashboard" "cost_projections" {
  title        = "[${var.target_env}] [${local.team_name}] ${local.app_name}"
  description  = "${var.target_env} - ${local.team_name} - ${local.app_name} dashboard."
  layout_type  = "ordered"
  is_read_only = "true"

  widget {
    group_definition {
      layout_type = "ordered"
      title       = "Kafka"

      widget {}
    }
  }
}

```

### Expected Behavior

A terraform plan is produced

### Actual Behavior

``` !  ~/R/o/losses-infra   cost-projects-in-cicd *  cost-projections/uat  terraform plan --var-file ../../_vars/uat.tfvars --var-file ../../_vars/global.tfvars                                                                            2733ms  Thu 16:39:02
datadog_monitor.kafka_monitor["tariffs-kafka-lag"]: Refreshing state... [id=18486560]
datadog_monitor.kube_monitor["available-pods"]: Refreshing state... [id=18486559]
datadog_monitor.kafka_monitor["personal-projections-kafka-lag"]: Refreshing state... [id=18486561]
datadog_monitor.kafka_monitor["contracts-kafka-lag"]: Refreshing state... [id=18486562]
datadog_dashboard.cost_projections: Refreshing state... [id=3nd-ahu-5py]

Error: rpc error: code = Unavailable desc = transport is closing

Error: rpc error: code = Unavailable desc = transport is closing

Error: rpc error: code = Unavailable desc = transport is closing

Error: rpc error: code = Unavailable desc = transport is closing

Error: rpc error: code = Unavailable desc = transport is closing
```

### Steps to Reproduce

Please list the steps required to reproduce the issue, for example:

1. Upgrade to 2.19.0
2. `terraform plan`

This is only happening with 2.19.0. If I downgrade the provider version to 2.18.1 it plans without problems.

### Important Factoids


### References

I also started taking a look and was able to reproduce with the following config:

```
# dashboard
resource "datadog_dashboard" "example_dashboard" {
  title        = "Example Ordered Dashboard"
  description  = "test dashboard"
  layout_type  = "ordered"
  is_read_only = "true"

  widget {
    group_definition {
      layout_type = "ordered"
      title       = "Exapmle Group Widget"

      widget {}
    }
  }
}
```

Running with `TF_LOG=trace terraform apply` shows:

```
Error: rpc error: code = Unavailable desc = transport is closing


panic: interface conversion: interface {} is nil, not map[string]interface {}
```

and points to `/resource_datadog_dashboard.go:768`

This looks to be because the widget definition inside the group widget is empty. 
@enicholson125 Can you run the your config with `TF_LOG=trace` and confirm if this is the exception you're seeing? I'm able to reproduce this behavior on both the latest and 2.18.1, so I want to ensure this is indeed the issue you're hitting.

--- a/datadog/resource_datadog_monitor.go
+++ b/datadog/resource_datadog_monitor.go
@@ -461,6 +461,10 @@ func resourceDatadogMonitorRead(d *schema.ResourceData, meta interface{}) error 
 		return err
 	}
 
+	if d.Id() == "" {
+		return nil
+	}
+
 	thresholds := make(map[string]string)
 
 	if v, ok := m.Options.Thresholds.GetOkOk(); ok {

